javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll

```diff
@@ -213,17 +213,24 @@ module TaintedPath {
         output = this
       )
       or
-      // non-global replace or replace of something other than /\.\./g
+      // non-global replace or replace of something other than /\.\./g, /[/]/g, or /[\.]/g.
       this.getCalleeName() = "replace" and
       input = getReceiver() and
       output = this and
-      not exists(RegExpLiteral literal, RegExpSequence seq |
+      not exists(RegExpLiteral literal |
         getArgument(0).getALocalSource().asExpr() = literal and
-        literal.isGlobal() and
-        literal.getRoot() = seq and
-        seq.getChild(0).(RegExpConstant).getValue() = "." and
-        seq.getChild(1).(RegExpConstant).getValue() = "." and
-        seq.getNumChild() = 2
+        literal.isGlobal()
+      |
+        exists(RegExpSequence seq | literal.getRoot() = seq |
+          seq.getChild(0).(RegExpConstant).getValue() = "." and
+          seq.getChild(1).(RegExpConstant).getValue() = "." and
+          seq.getNumChild() = 2
+        )
+        or
+        exists(RegExpCharacterClass choice | literal.getRoot() = choice |
+          choice.getAMatchedString() = "/" or
+          choice.getAMatchedString() = "."
```

Comment:
Do we need to restrict this to character classes? If we declared `choice` to be `RegExpTerm` instead it seems like it would work for `/\./g` and `/\//g` as well.

It's a bit sad that we block normalized absolute paths here. Wouldn't it be worth adding that as a step?

Reply:
> Do we need to restrict this to character classes? If we declared choice to be RegExpTerm instead it seems like it would work for /\./g and /\//g as well.

:+1: 
The same works for the existing `RegExpSequence` in the predicate.

Reply:
> It's a bit sad that we block normalized absolute paths here. Wouldn't it be worth adding that as a step?

I think it could be.  
I added a taint-step for replace calls that remove only dots, and not forward slash.